napalm/junos/junos.py

```diff
@@ -1106,6 +1106,18 @@ def build_prefix_limit(**args):
                 # we do not want cluster in the output
                 del bgp_config[bgp_group_name]['cluster']
 
+            # Fix local autonomous-system (local_as) in each BGP group
+            # if local-as is 0 in the BGP group means we are not using local-as comand
+            # but the AS number from routing-options/autonomous-system
+            bgp_as = junos_views.junos_routing_options_autonomous_system(self.device)
```

Comment:
This clearly should be outside of this for loop. It's pointless to execute this RPC for each and every BGP group, please move it.